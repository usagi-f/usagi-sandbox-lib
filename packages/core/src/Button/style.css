.button {
  background-color: red;
  color: var(--mono50);
  font-size: 24px;
}

.button[data-theme="dark"] {
  background-color: darkred;
}
