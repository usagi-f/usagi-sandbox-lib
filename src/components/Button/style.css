.button {
  background-color: red;
  color: var(--mono50);
  font-size: 24px;
}
