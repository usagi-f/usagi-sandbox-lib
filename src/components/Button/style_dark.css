.button {
  background-color: darkred;
  border-radius: 5px;
  color: var(--mono50);
  font-size: 12px;
  padding: 5px 20px;
}
