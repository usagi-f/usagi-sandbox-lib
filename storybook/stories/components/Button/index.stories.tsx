import * as React from 'react';
import { storiesOf } from '@storybook/react';

import { Button as ButtonComponent, IButtonProps } from 'usagi-sandbox-lib';
import 'usagi-sandbox-lib/dist/css/all.css';

export const Button: React.FC<IButtonProps> = props => <ButtonComponent {...props} />;

storiesOf('Button', module)
  .add('Simple render', () => <Button />)
  .add('with children', () => <Button>Hello Button</Button>)
  .add('with text', () => <Button text="Custom Props Text" />);
