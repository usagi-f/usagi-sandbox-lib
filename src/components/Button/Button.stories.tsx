import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { Button } from './Button';

storiesOf('Button', module)
  .add('Simple render', () => <Button />)
  .add('with children', () => <Button>Hello Button</Button>)
  .add('with text', () => <Button text="Custom Props Text" />);
