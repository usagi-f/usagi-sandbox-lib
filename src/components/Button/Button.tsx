import * as React from 'react'
import * as style from './style.css'
import { Context } from '../../context'

interface IButtonProps extends React.Props<{}> {
  text?: string;
}

const Button: React.FC<IButtonProps & Partial<Context>> = props => (
  <button
    type="button"
    data-theme={props.theme}
    className={style.button}
  >
    {props.children || props.text}
  </button>
)

Button.defaultProps = {
  text: 'MyButton',
}

export {
  Button,
  IButtonProps,
}
