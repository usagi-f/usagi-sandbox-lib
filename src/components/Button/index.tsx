import * as React from 'react'
import * as style from './style.css'

interface IButtonProps extends React.Props<{}> {
  text?: string;
}

const Button: React.FC<IButtonProps> = (props: IButtonProps) => (
  <button type="button" className={style.button}>{props.children || props.text}</button>
)

Button.defaultProps = {
  text: 'MyButton',
}

export {
  Button,
  IButtonProps,
}
