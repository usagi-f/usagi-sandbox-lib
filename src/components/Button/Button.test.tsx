import * as React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { Button, IButtonProps } from './Button';

test('Rendered by props', () => {
  const props: IButtonProps = {
    text: 'foo',
  };
  const wrapper = shallow(<Button {...props} />);
  expect(wrapper.text()).toBe(props.text);
});

test('Snapshot', () => {
  const props: IButtonProps = {
    text: 'foo',
  };
  const tree = shallow(<Button {...props} />);
  expect(toJson(tree)).toMatchSnapshot();
});
