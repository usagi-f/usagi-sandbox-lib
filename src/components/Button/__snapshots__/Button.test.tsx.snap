// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
<button
  type="button"
>
  foo
</button>
`;
